src/components/ModalUnregisteredTokenInfo.js

```diff
@@ -26,21 +32,41 @@ class ModalUnregisteredTokenInfo extends React.Component {
    * errorMessage {String} Message to show in case of an error when registering the token
    * formValidated {boolean} If register form was already validated
    */
-  state = { token: null, errorMessage: '', formValidated: false };
+  state = {
+    errorMessage: '',
+    formValidated: false,
+    totalSupply: 0,
+    canMint: false,
+    canMelt: false,
+    transactionsCount: 0,
+  };
 
   // Reference to the form
   form = React.createRef();
 
   componentDidMount() {
+    $('#unregisteredTokenInfoModal').modal('show');
+
     $('#unregisteredTokenInfoModal').on('hidden.bs.modal', () => {
       this.setState({ errorMessage: '', formValidated: false });
+      this.props.onClose();
     });
+
+    this.updateTokenInfo();
   }
 
-  componentDidUpdate = (prevProps) => {
-    if (prevProps.token !== this.props.token) {
-      this.setState({ token: this.props.token });
-    }
+  async updateTokenInfo() {
```

Comment:
We would need to duplicate this method in every component we want to use the `TokenGeneralInfo`? Why don't we move this to that component, so the logic is fully there and we only use the component?

Reply:
I understood your refactor but we still need to duplicate code every time we want to use `TokenGeneralInfo`, right? Why don't we move `updateTokenInfo` to it, so we remove this code repetition?

Reply:
Because, again, `TokenGeneralInfo` is a `component`, not a screen, I don't think it should have side effects and should only be responsible for rendering what is passed as props

I think we should refactor the `tokenInfo` fetching mechanism to be a saga and just dispatch `fetchTokenInfo(tokenId)` actions, caching responses on the redux state (and not re-fetching if they are already there)

I would just suggest doing this refactor in a future PR, and continue with the code duplicates until then instead of dirtying the `TokenGeneralInfo` component with side effects